### ISSUE TITLE
Dock hides despite mouse hovering over it

### ISSUE THREAD

**Issue by balderclaassen** [passingthru67/workspaces-to-dock#115]
###Linux Distribution version
Ubuntu Gnome 17.04

###Gnome Shell version
3.24.2

###Extension version or branch
42

###Description of the problem
The dock hides even if my mouse cursor is on it.
The dock is set to autohide, pressure reveals it. When I move the cursor outside the dock, and back inside the dock before it hides I expect the hide timer to reset and the dock to stay in view. 
Currently the dock hides, despite my mouse being inside of it.

###Steps to reproduce the problem
Autohide should be on, timing options for easy reproduction would be animation low (50ms) and hide delay something high (3000ms).
Move the mouse to the edge of the screen to reveal the dock, move it outside the dock and back in before it hides.

I have looked around in the code to see if I could find what needs to be done to achieve this, but I am rather new to gnome extensions (and javascript to be frank) so I haven't been able to figure it out yet. 

Thanks for reading.

**Comment by passingthru67** [comment]
This sounds like the same problem as issue #95 (at least similar).
Please download the 'fix-show-hide-animation' branch and let me know if the problem goes away.

**Comment by balderclaassen** [comment]
I have read #95 and tried out the other branch.
As far as being related to #95, I do not have "Disable scroll when dock is hidden to prevent workspace switching" turned on (nor does turning it off help btw) And I have no problems in getting it to show. It does appear when applying pressure. 
The 'fix-show-hide-animation' branch did not fix the issue. It did however break the hide delay. With this branch on there is no hide-delay anymore. I can set it to 1500, I can set it to 250, it just hides instantly.

I also tried to install the latest master from github but that seems to behave the same as what I have from the Gnome-extensions website.

**Comment by passingthru67** [comment]
Thanks for the dconf dump. That will help me duplicate your setup. BTW, do you have any other extensions installed?

**Comment by balderclaassen** [comment]
These are the extensions I have activated:
Advanced Volume Mixer  
AlternateTab  
AppKeys  
Auto Move Windows  
Calculator  
Coverflow Alt-Tab  
Drop Down Terminal   
Extensions  
gTile 
Hide Top Bar 
Hot Edges  
Impatience 
Night Light Slider  
No Topleft Hot Corner  
Pixel Saver  
Pomodoro  
Removable Drive Menu  
Scroll Panel  
Show Desktop from Overview  
TopIcons Plus  
Web Search Dialog  
Web Search From Clipboard  
windowNavigator  by fmuellner
WindowOverlay Icons  
System extension
Workspaces to Dock 

Impatience is the only one I could think of that might be interrupting it but with it turned off and a refresh didn't change anything.

**Comment by balderclaassen** [comment]
With every gnome extension turned off and only workspaces-to-dock turned on and after a ALT+F2 r the behavior is the same.

**Comment by balderclaassen** [comment]
I have narrowed it down to one of the settings. At least so I assume. I reset the dconf to defaults and it works fine. so now I am gonna reset it to my settings one by one and see when the problem occurs.
Will edit the results in this comment.

**Comment by passingthru67** [comment]
Thanks for testing things out. The information will help as I work on a fix for the problem. In the meantime, I'm glad you found a workaround.

**Comment by passingthru67** [comment]
@balderclaassen, I've committed a new branch that I hope fixes the dock show/hide issues when animation timing is adjusted or turned off. Please download the 'fix-show-hide' branch and let me know if the problem goes away.

**Comment by balderclaassen** [comment]
It works great ! :+1: 
Thank you very much, the workaround was fine, but I definetly prefer this faster animation. I also turned Impatience back on (its main use is showing the dash quicker) and it works great with that as well. My setting is now 35/400. I also tested a bunch of others, high hide timings, and animation as low or 0 or 10. It all works as expected.
Thanks for taking the time to fix it, it's appreciated ^^
